examples/io_perf_test.cpp

```diff
@@ -0,0 +1,176 @@
+    {
+        print_version_map();
+    }
+
+    if (argc < 2) 
+    {
+        std::cerr << "usage: otio_io_perf_test path/to/timeline.otio";
+        std::cerr << std::endl;
+        return 1;
+    }
+
+    otio::ErrorStatus err;
+
+    otio::schema_version_map downgrade_manifest = {
+        {"FakeSchema", 3},
+        {"Clip", 1},
+        {"OtherThing", 12000}
+    };
+
+    if (RUN_STRUCT.CLONE_TEST)
+    {
+        otio::SerializableObject::Retainer<otio::Clip> cl = new otio::Clip("test");
+        cl->metadata()["example thing"] = "banana";
+        const auto intermediate = cl->clone(&err);
+        assert(intermediate != nullptr);
+        const auto cl_clone = dynamic_cast<otio::Clip*>(intermediate);
+        assert(cl_clone != nullptr);
+        assert(!otio::is_error(err));
+        assert(cl->name() == cl_clone->name());
+    }
+
+    if (RUN_STRUCT.SINGLE_CLIP_DOWNGRADE_TEST)
+    {
+        otio::SerializableObject::Retainer<otio::Clip> cl = new otio::Clip("test");
+        cl->metadata()["example thing"] = "banana";
+        chrono_time_point begin = std::chrono::steady_clock::now();
+        cl->to_json_file("/var/tmp/clip.otio", &err, &downgrade_manifest);
+        chrono_time_point end = std::chrono::steady_clock::now();
+        print_elapsed_time("downgrade clip", begin, end);
+    }
+
+    otio::any tl;
+    std::string fname = std::string(argv[1]);
+
+    chrono_time_point begin = std::chrono::steady_clock::now();
+    otio::SerializableObject::Retainer<otio::Timeline> timeline(
+            dynamic_cast<otio::Timeline*>(
+                otio::Timeline::from_json_file(argv[1], &err)
+            )
+    );
+    chrono_time_point end = std::chrono::steady_clock::now();
+    if (!timeline)
+    {
+        examples::print_error(err);
+        return 1;
+    }
+
+    print_elapsed_time("deserialize_json_from_file", begin, end);
+
+
+    if (RUN_STRUCT.TO_JSON_STRING)
+    {
+        begin = std::chrono::steady_clock::now();
+        const std::string result = timeline.value->to_json_string(
+                &err,
+                &downgrade_manifest
+        );
+        end = std::chrono::steady_clock::now();
+
+        if (otio::is_error(err))
+        {
+            examples::print_error(err);
+            return 1;
+        }
+        print_elapsed_time("serialize_json_to_string", begin, end);
+    }
+
+    if (RUN_STRUCT.TO_JSON_STRING_NO_DOWNGRADE)
+    {
+        begin = std::chrono::steady_clock::now();
+        const std::string result = timeline.value->to_json_string(&err, {});
+        end = std::chrono::steady_clock::now();
+
+        if (otio::is_error(err))
+        {
+            examples::print_error(err);
+            return 1;
+        }
+        print_elapsed_time("serialize_json_to_string [no downgrade]", begin, end);
+    }
+
+    if (RUN_STRUCT.TO_JSON_FILE)
+    {
+        begin = std::chrono::steady_clock::now();
+        timeline.value->to_json_file(
+                "/var/tmp/io_perf_test.otio",
```

Comment:
In the "util.h" header file there is a function "std::string create_temp_dir();" that creates a temporary directory in a portable way.